emuflight/EmuFlight
# Emuflight 1.0.0 master rust

Here's a pull request for how to get started with Rust Integration in EmuFlight 1.0.0 . The code size is still slightly higher (~1kb) But overall, the integration was smooth. The idea is we move more and more functions to the rustemu crate (and delete them from the C codebase)

To test this out,

 * Make sure you have Rust installed ( https://www.rust-lang.org/tools/install )
```
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
* Restart the terminal to make sure your PATH variable can use the rustup command and make sure to install the bare metal arm toolchain for rust installed:

```
rustup target add thumbv7em-none-eabihf
```
* Then you can build emuflight using the regular Makefile
```
make TARGET=STM32F411
```
Let me know if there are any issues, so I can fix them before this is merged.
